### ISSUE TITLE
Ability to set StopOnError via .runsettings

### ISSUE THREAD

**Issue by queil** [nunit/nunit3-vs-adapter#675]
It looks (both from documentation and code) that StopOnError is not supported. Are there any plans to support it?

**Comment by OsirisTerje** [comment]
You're right it is not supported, and currently there are no plans.  There haven't been much asking for supporting this either, so I think this is a first :-)  

I can't see why I can't be supported, so let us see if this gets some more votes :-)

**Comment by OsirisTerje** [comment]
@queil 
3 up-votes are awesome, more than enough.   

And, it is done!  

Please check the dev build on https://www.myget.org/feed/nunit/package/nuget/NUnit3TestAdapter  (One on or later than Jan 29th)

To enable it in runsettings:

```
<NUnit>
    <StopOnError>true</StopOnError>
</NUnit>
```

**Comment by queil** [comment]
@OsirisTerje Thanks for that!

**Comment by queil** [comment]
@OsirisTerje Hi, so we did some initial tests. Passing the param works fine. However, stopping on error itself only works for `ParallelScope.None`. Both `ParallelScope.Children` and `ParallelScope.Fixtures` seem to ignore the setting. Is that NUnit's or the VS Adapter's issue?

**Comment by OsirisTerje** [comment]
Can you guys chime in on this one @rprouse @jnm2 @ChrisMaddock ? I would guess this is NUnit itself, since the adapter now only pass the parameter further down.  
@queil  Have you checked with nunit.console ?  That is doing the same thing, so it should show the same behavior.

**Comment by queil** [comment]
@OsirisTerje I haven't checked with nunit.console yet but I will try when I get a chance.

**Comment by ChrisMaddock** [comment]
I’m not sure of the implementation of this one without checking, sorry. I suspect the problem will lie in the engine or framework, it’s not been a heavily used option up till now.

Interested to hear what you find, @queil

**Comment by queil** [comment]
@ChrisMaddock @OsirisTerje So just tested it and nunit.console (https://github.com/nunit/nunit-console/releases/tag/v3.11) behaves the same way. I.e. stop on error is ignored when running tests in parallel. I guess it is by design?

**Comment by ChrisMaddock** [comment]
Either 'by design' or 'by error'. 😄 

I had a quick look, looks like the framework doesn't have any functionality to take into account tests that might be running in parallel. My opinion, is that that should be corrected - although I don't know the framework codebase well enough to know how it should be structured.

It would be pretty easy to do through the engine. Maybe we should move this feature up a level...

**Comment by OsirisTerje** [comment]
@ChrisMaddock 
1) Yes, would be nice to have this in the next version
2) Added this issue in NUnit Console/Engine : https://github.com/nunit/nunit-console/issues/734

**Comment by OsirisTerje** [comment]
@queil  Beta released on Nuget, see https://www.nuget.org/packages/NUnit3TestAdapter/3.17.0-beta.1 . Please check that it works :-)
